From 8e28892cc7da99f9d26eb6cc22c5c3c196d5e07a Mon Sep 17 00:00:00 2001
From: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Date: Wed, 17 Feb 2016 02:10:52 +0200
Subject: [PATCH 45/94] tpm_crb/tis: fix: use dev_name() for /proc/iomem

commit 30f9c8c9e2ea37473a51354e9e492580a40661ce from upstream

In all cases use dev_name() for the mapped resources. This is both
for sake of consistency and also with some platforms resource name
given by ACPI object seems to return garbage.

Signed-off-by: Jarkko Sakkinen <jarkko.sakkinen@linux.intel.com>
Fixes: 1bd047be37d9 ("tpm_crb: Use devm_ioremap_resource")
---
 drivers/char/tpm/tpm_crb.c |    4 +++-
 drivers/char/tpm/tpm_tis.c |    4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/char/tpm/tpm_crb.c b/drivers/char/tpm/tpm_crb.c
index 916332c..9eb0404 100644
--- a/drivers/char/tpm/tpm_crb.c
+++ b/drivers/char/tpm/tpm_crb.c
@@ -227,8 +227,10 @@ static int crb_check_resource(struct acpi_resource *ares, void *data)
 	struct crb_priv *priv = data;
 	struct resource res;
 
-	if (acpi_dev_resource_memory(ares, &res))
+	if (acpi_dev_resource_memory(ares, &res)) {
 		priv->res = res;
+		priv->res.name = NULL;
+	}
 
 	return 1;
 }
diff --git a/drivers/char/tpm/tpm_tis.c b/drivers/char/tpm/tpm_tis.c
index ca137b5..2b2eff9 100644
--- a/drivers/char/tpm/tpm_tis.c
+++ b/drivers/char/tpm/tpm_tis.c
@@ -930,8 +930,10 @@ static int tpm_check_resource(struct acpi_resource *ares, void *data)
 
 	if (acpi_dev_resource_interrupt(ares, 0, &res))
 		tpm_info->irq = res.start;
-	else if (acpi_dev_resource_memory(ares, &res))
+	else if (acpi_dev_resource_memory(ares, &res)) {
 		tpm_info->res = res;
+		tpm_info->res.name = NULL;
+	}
 
 	return 1;
 }
-- 
1.7.5.4

